kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/StampViewModel2.java

```diff
@@ -36,23 +66,94 @@ public enum StampProperties {
         addProperty(PATHS, Collections.emptyList(), true);
         addProperty(STATUSES, Collections.emptyList(), true);
+
+        this.eventBus = EvtBusFactory.getDefaultEvtBus();
     }
 
-    @Override
-    public StampViewModel save(boolean force) {
-        return (StampViewModel) super.save(force);
+    public void init(EntityFacade entity, UUID topic, ViewProperties viewProperties) {
+        this.entityFacade = entity;
+        this.viewProperties = viewProperties;
+        this.topic = topic;
+
+        // listen to events that the properties panel is going to be closed
+        closePropertiesPanelEventSubscriber = evt -> onPropertiesPanelClose();
+        eventBus.subscribe(topic, ClosePropertiesPanelEvent.class, closePropertiesPanelEventSubscriber);
+
+        // initialize observable lists
+        setPropertyValues(MODULES, fetchDescendentsOfConcept(viewProperties, TinkarTerm.MODULE.publicId()));
+        setPropertyValues(PATHS, fetchDescendentsOfConcept(viewProperties, TinkarTerm.PATH.publicId()));
+        setPropertyValues(STATUSES, List.of(State.values()));
+
+        loadStamp();
+        loadStampValuesFromDB();
+
+        doOnChange(this::validate, STATUS, MODULE, PATH);
     }
 
+    private void onPropertiesPanelClose() {
+        loadStampValuesFromDB(); // every time the properties panel closes we reset the form
+    }
+
+    private void loadStamp() {
+        EntityVersion latestVersion = viewProperties.calculator().latest(entityFacade).get();
+        StampEntity stampEntity = latestVersion.stamp();
+
+        setPropertyValue(StampProperties.CURRENT_STAMP, stampEntity);
+    }
+
+    private void loadStampValuesFromDB() {
+        StampEntity stampEntity = getPropertyValue(StampProperties.CURRENT_STAMP);
+
+        setPropertyValue(StampViewModel2.StampProperties.STATUS, stampEntity.state());
+        setPropertyValue(StampViewModel2.StampProperties.MODULE, stampEntity.module());
+        setPropertyValue(PATH, stampEntity.path());
+    }
+
+    private boolean updateIsStampValuesChanged() {
+        StampEntity stampEntity = getPropertyValue(StampProperties.CURRENT_STAMP);
+
+        boolean same = stampEntity.state() == getPropertyValue(StampViewModel2.StampProperties.STATUS)
+                && stampEntity.path() == getPropertyValue(PATH)
+                && stampEntity.module() == getPropertyValue(MODULE);
+
+        setPropertyValue(IS_STAMP_VALUES_THE_SAME, same);
+
+        return same;
+    }
+
+    public void cancel(Node eventSource) {
+        eventBus.publish(topic, new ClosePropertiesPanelEvent(eventSource,
+                ClosePropertiesPanelEvent.CLOSE_PROPERTIES));
+    }
+
+    public void reset(Node eventSource) {
+        ConfirmationDialogController.showConfirmationDialog(eventSource, CONFIRM_CLEAR_TITLE, CONFIRM_CLEAR_MESSAGE)
+            .thenAccept(confirmed -> {
+                if (confirmed) {
+                    loadStampValuesFromDB();
```

Comment:
Assuming you've called `save(true)` during init(...) method, the reset should be able to work.
Also, make sure properties bound to UI controls are bidirectionally bound (within the initialize() method in the controller). When reset occurs (copy model values into the property values the UI will change).

Reply:
@carldea I can and I tried it but reset still didn't work. I'll add the save(true) anyway

Yes. The UI controls are bidirectionally bound in the controller code

P.S.: all this functionality is working correctly... just not using reset() at the moment.